app/routers/organizations.py

```diff
@@ -1,15 +1,43 @@
 from fastapi import APIRouter, status, HTTPException
 from database import client
 from models import Organization
+from fastapi.responses import JSONResponse
+import string
+import random
 
 router = APIRouter()
 
 
+@router.post("/create-org/{org_name}")
+async def create_organization(org_name: str):
+    if org_name is not None:
+        # create an API key
+        char_set = string.ascii_letters + string.punctuation
+        urand = random.SystemRandom()
+        org_key = "".join([urand.choice(char_set) for _ in range(20)])
+
+        # check if API key exists
+        if (client.quiz.organization.find_one({"org_key": org_key})) is None:
```

Comment:
Why no check for `org_name`? 

If you don't want to check for the name: since the key is generated here and not passed by the user, it doesn't make sense to return an error if the key exists. Why not just try with a new random key?

Reply:
There is a check for `org_name`. 

I thought of the same thing, but I was worried about what if we reach a case where it enters an infinite loop. Won't be happening in the near future, but still.

Reply:
Add the loop with a max of 3 iterations. Break as soon as a unique id is found. No reason here to return an error when it wasn’t the user's fault

Also, use the same secrets library that Plio uses.